core/src/main/antlr4/org/apache/druid/math/expr/antlr/Expr.g4

```diff
@@ -51,9 +51,12 @@ numericElement : (LONG | DOUBLE | NULL);
 literalElement : (STRING | LONG | DOUBLE | NULL);
 
 NULL : 'null';
+LONG : [0-9]+;
+EXP: [eE] [-]? LONG;
+// DOUBLE provides partial support for java double format
+// see: https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#valueOf-java.lang.String-
+DOUBLE : 'NaN' | 'Infinity' | (LONG '.' LONG) | (LONG EXP) | (LONG '.' LONG EXP);
```

Comment:
This used to allow `10.` as a double, but now it doesn't. I think we should add that back. (with tests 🙂)

Reply:
good catch :+1:

fixed, and added tests